ecl/hql/hqlgram.y

```diff
@@ -4256,6 +4268,39 @@ recordDef
                             $$.setExpr(LINK(record));
                             $$.setPosition($1);
                         }
+     | RECORDOF '(' dataSet ',' recordDef ',' lookupOption optOpt ')'
+                        {
+                            OwnedHqlExpr name = createValue(no_nameof, makeStringType(UNKNOWN_LENGTH, NULL, NULL), $3.getExpr());
+                            $3.release().setExpr(name.getClear());
+                            parser->normalizeExpression($3, type_string, true);
+                            IHqlExpression * record = parser->lookupDFSlayout($1, $3.getExpr(), $5.getExpr(), $7.getExpr(), $8.getBoolValue());
+                            $3.clear();
+                            $$.setExpr(record);
+                            $$.setPosition($1);
+                        }
+    | RECORDOF '(' dataSet ',' lookupOption ')'
+                        {
+                            OwnedHqlExpr name = createValue(no_nameof, makeStringType(UNKNOWN_LENGTH, NULL, NULL), $3.getExpr());
+                            $3.release().setExpr(name.getClear());
+                            parser->normalizeExpression($3, type_string, true);
+                            IHqlExpression * record = parser->lookupDFSlayout($1, $3.getExpr(), LINK(queryNullRecord()), $5.getExpr(), false);
```

Comment:
You could argue that in this case the default layout should be taken from the dataset $3 rather than using queryNullRecord(). @ghalliday What do you think?